engine/table/src/test/java/io/deephaven/engine/table/impl/CapturingUpdateGraph.java

```diff
@@ -0,0 +1,153 @@
+package io.deephaven.engine.table.impl;
+
+import io.deephaven.base.log.LogOutput;
+import io.deephaven.engine.context.ExecutionContext;
+import io.deephaven.engine.testutil.ControlledUpdateGraph;
+import io.deephaven.engine.updategraph.LogicalClock;
+import io.deephaven.engine.updategraph.NotificationAdapter;
+import io.deephaven.engine.updategraph.UpdateGraph;
+import io.deephaven.io.log.LogEntry;
+import io.deephaven.util.SafeCloseable;
+import io.deephaven.util.function.ThrowingRunnable;
+import io.deephaven.util.locks.AwareFunctionalLock;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+final class CapturingUpdateGraph implements UpdateGraph {
+
+    final ControlledUpdateGraph delegate;
+
+    final ExecutionContext context;
```

Comment:
Might be nice to keep these private and add getters, if this will be a shared utility. Also, maybe we want to make this class and its constructor public for general test use?